### PR TITLE
Fix for surface meshes

### PR DESCRIPTION
**Pull request by stefanozampini**
You may want to include this small fix before the release.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1323](https://github.com/mfem/mfem/pull/1323) | @stefanozampini | @tzanio | @tzanio + @v-dobrev | 02/27/20 | 02/28/20 | 02/29/20 | |

### CODE REVIEW

**Comment by jandrej** [comment]
@stefanozampini Can you briefly describe what the issue was before?

**Comment by jandrej** [comment]
@stefanozampini Can you briefly describe what the issue was before?

**Comment by stefanozampini** [comment]
The old code was using  GetDim from the finite element, while the shape functions are vector valued functions in space dimension

File: fem/bilininteg.cpp [hunk 1]
```diff
@@ -1845,19 +1844,20 @@ void VectorFEMassIntegrator::AssembleElementMatrix2(
        && trial_fe.GetRangeType() == FiniteElement::VECTOR)
    {
       // assume test_fe is scalar FE and trial_fe is vector FE
-      int dim  = test_fe.GetDim();
+      int dim = test_fe.GetDim();
+      int spaceDim = Trans.GetSpaceDim();
```
Comment (v-dobrev) [inline fem/bilininteg.cpp]: I'm not sure this whole section makes sense when `dim != spaceDim` but I still see some uses of `dim` that need to be converted to `spaceDim`. For example, the number of components in the scalar space should be `spaceDim` and in particular the element matrix should be `spaceDim*test_dof x trial_dof`. Also, the loops from `0` to `dim-1` should go up to `spaceDim-1`.

One exception is the case of matrix coefficient -- then the scalar space must have as many components as the number of rows in the matrix coefficient.

Comment (stefanozampini) [reply fem/bilininteg.cpp]: Yeah, Do you have a testing code for that? I have only noticed the issue with both trial and test being vector spaces. If you know what needs to be fixed, go ahead and push here

Comment (v-dobrev) [reply fem/bilininteg.cpp]: Done.

**Comment by tzanio** [comment]
Merged in `next` for testing...

File: fem/bilininteg.cpp (hunk 1 above)
Comment (stefanozampini) [reply fem/bilininteg.cpp]: thanks

**Pull request merged**